internal/pkg/agent/cmd/diagnostics.go

```diff
@@ -639,16 +642,77 @@ func saveLogs(name string, logPath string, zw *zip.Writer) error {
 
 // writeFile writes json or yaml data from the interface to the writer.
 func writeFile(w io.Writer, outputFormat string, v interface{}) error {
+	redacted, err := redact(v)
+	if err != nil {
+		return err
+	}
+
 	if outputFormat == "json" {
 		je := json.NewEncoder(w)
 		je.SetIndent("", "  ")
-		return je.Encode(v)
+		return je.Encode(redacted)
 	}
+
 	ye := yaml.NewEncoder(w)
-	err := ye.Encode(v)
+	err = ye.Encode(redacted)
 	return closeHandlers(err, ye)
 }
 
+func redact(v interface{}) (redacted map[string]interface{}, err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("failed redacting data: %w", err)
+		}
+	}()
+	redacted = map[string]interface{}{}
+	bs, err := yaml.Marshal(v)
+	if err != nil {
+		return nil, err
+	}
+
+	err = yaml.Unmarshal(bs, &redacted)
+	if err != nil {
+		return nil, err
+	}
+
+	return redactMap(redacted), nil
+}
+
+func toMapStr(v interface{}) map[string]interface{} {
+	mm := map[string]interface{}{}
+	m, ok := v.(map[interface{}]interface{})
+	if !ok {
+		return mm
+	}
+
+	for k, v := range m {
+		mm[k.(string)] = v
+	}
+	return mm
+}
+
+func redactMap(m map[string]interface{}) map[string]interface{} {
+	for k, v := range m {
+		if v != nil && reflect.TypeOf(v).Kind() == reflect.Map {
+			v = redactMap(toMapStr(v))
+		}
+		if redactKey(k) {
+			v = REDACTED
+		}
+		m[k] = v
+	}
+	return m
+}
+
+func redactKey(k string) bool {
+	return k != "routekey" &&
```

Comment:
Always go for explicit over single statement or returned values, because it reduce the cognitive load, the compile take care of optimization.

```golang
//  "routekey" is an internal value use to differentiate multiple running configuration.
if k == "routekey" {
return false
}
...